sqlmesh/core/state_sync/base.py

```diff
@@ -168,6 +189,60 @@ def missing_intervals(
                 missing[snapshot] = intervals
         return missing
 
+    def get_versions(self, validate: bool = True) -> t.Optional[t.Tuple[int, str]]:
+        """Get the current versions of SQLMesh and libraries.
+
+        Args:
+            validate: Whether or not to raise error if the running version is ahead of state.
+
+        Returns:
+            The row containing version information.
+        """
+        row = self._get_versions()
+
+        def raise_error(
+            lib: str, local: str | int, remote: str | int, migrate: bool = False
+        ) -> None:
+            if migrate:
```

Comment:
Why is this called `migrate`? As far as I understand it just determines whether to throw an `ahead` or `behind` version of the error.